src/components/admin/lib/__tests__/article-validators.test.js

```diff
@@ -0,0 +1,87 @@
+import { hasNonHttpsURL, returnsFirstRelativeURL } from '../article-validators';
+
+describe('hasNonHttpsURL', () => {
+  it('Returns true if it finds an http (not s) link in an href', () => {
+    expect(
+      hasNonHttpsURL('<a href=\'http://www.w3schools.com\'> Title\'hello\'</a>')
+    ).toBe(true);
+    expect(
+      hasNonHttpsURL(
+        '<a href=\'http://www.w3schools.com/\' target=\'_blank\'>Visit W3Schools!</a>'
+      )
+    ).toBe(true);
+  });
+  it('Returns false if it finds an https in an href', () => {
+    expect(
+      hasNonHttpsURL('<a href=\'https://www.w3schools.com\'> Title\'hello\'</a>')
+    ).toBe(false);
+  });
+  it('Returns true if it finds an http (not s) link in an img tag', () => {
+    expect(
+      hasNonHttpsURL('<img src=\'http://image.com\' alt=\'Mountain View\'>')
+    ).toBe(true);
+    expect(
+      hasNonHttpsURL(
+        '<img src=\'http://image.com\' alt=\'HTML tutorial\' style=\'width:42px;height:42px;border:0;\'>'
+      )
+    ).toBe(true);
+  });
+  it('Returns false if it finds an https in an img tag', () => {
+    expect(
+      hasNonHttpsURL('<img src=\'https://image.com\' alt=\'Mountain View\'>')
+    ).toBe(false);
+  });
+  it('Returns false when encountering a url in the text that is not a link', () => {
+    expect(hasNonHttpsURL('<p>http://www.google.com</p>')).toBe(false);
+    expect(
```

Comment:
At first I thought this test was a mistake as I didn't think there were any links not in text here, but now I see, and I actually really love this test, that's a pretty dangerous edgecase, good job @mdmjg !